### DISCUSSION TITLE
Dynamically change MenuItem name

### DISCUSSION THREAD

**Discussion by vzahradnik** [davetcc/tcMenu#379] [Q&A] [answered]
I have the following usecase:
- Defined a submenu called e.g. "Temperature sensors" with menu items "Sensor 1", "Sensor 2", ...
- These items show current sensor temperatures in degrees Celsius

I'd like to customize menu item names, e.g. rename "Sensor 1" to "Sensor (top)", ... but only when some conditions are met. I support multiple usecases in a single firmware and I'd like to make this as flexible as possible.

Is it possible to change the actual item name programatically? As an alternative, I could probably define RuntimeMenuItems but besides the name change, I'd like to keep everything as-is, including the EEPROM address.

Any hint is much appreciated.

Thanks!

**Comment by davetcc** [comment, marked as answer]
See the dynamicMenuItem example and the core menu item docs, they both describe how to do this. Basically it comes down to choosing the info block in RAM option in designer for those items.

**Reply by vzahradnik** [reply]
I found the RAM option in the designer so that was my guess. Thanks for pointing out.

**Reply by davetcc** [reply]
![image](https://github.com/davetcc/tcMenu/assets/12195465/069202a2-06dc-40b2-b1f7-b97eb5784c9a)

Actually I did mean to add this image a couple of days ago. This is the option.

**Reply by vzahradnik** [reply]
I thought so. Thanks for the picture anyway.

**Comment by vzahradnik** [comment]
All is working perfectly. I let TcMenu generate code and then I figured out what to do. Your example just confirmed that I replace the string correctly.

**Comment by vzahradnik** [comment]
@davetcc what about project name? It is shown in the top status bar and I'd like to customize its title as well. Is something like this supported?

For context, I build a firmware with several flavors. These are part of the same codebase and I apply customizations based on flavor definition. So if there is one flavor, I'll enable some submenus, hide others, apply different display strings etc. Changing project name to flavor name is the last thing I need. It's a nice to have feature.

Thanks!
![image](https://github.com/davetcc/tcMenu/assets/4460766/7eedc8a3-d686-4187-bddb-e2a64fb9e696)

**Reply by davetcc** [reply]
Yes take a look at the pico example below. You just start the application name with a % and then manually add entries into the properties file. It works as expected although maybe not entirely polished.

https://github.com/davetcc/tcMenuLib/tree/master/examples/arduino32/picoAdafruitDashboard
